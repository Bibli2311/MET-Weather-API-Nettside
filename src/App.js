import React, { useState, useEffect } from "react";

function App() {
  const [nodeContent, setNodeContent] = useState([]);

  useEffect(() => {
    fetch("https://api.met.no/weatherapi/metalerts/1.1?show=all")
      .then((resp) => resp.text())
      .then((result) => {
        let xmlData = new window.DOMParser().parseFromString(result, "text/xml");

        let titleArray = xmlData.getElementsByTagName("item");

        let nodes = [];
        let elem = titleArray[0];
        elem.childNodes.forEach((node) => {
          nodes.push(node.textContent);
        });
        setNodeContent(nodes);
      });
  }, []);

  return (
    <div>
      {nodeContent.map((content, index) => (
        <p key={index}>{content}</p>
      ))}
    </div>
  );
}

export default App
