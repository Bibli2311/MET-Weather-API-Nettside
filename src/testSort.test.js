import renderer from 'react-test-renderer'
import App from './App'

it("Test if data is sorted correctly", () =>
{
    const component = renderer.create(
        <App/>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})