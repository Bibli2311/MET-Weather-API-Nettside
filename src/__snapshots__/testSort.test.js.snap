// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if data is sorted correctly 1`] = `<div />`;
